docker/tarfile/src.go

```diff
@@ -249,6 +249,9 @@ func (s *Source) ensureCachedDataIsPresentPrivate() error {
 	if err := json.Unmarshal(configBytes, &parsedConfig); err != nil {
 		return errors.Wrapf(err, "Error decoding tar config %s", tarManifest[0].Config)
 	}
+	if parsedConfig.RootFS == nil {
+		return errors.Errorf("Invalid image config %s, rootFS is not set", tarManifest[0].Config)
```

Comment:
Maybe we can move `%s` at the end of the error message? "rootFS is not set" is the important part but it might be hard to catch that as the config string might be long.

Reply:
The value is a path within the tar archive, the JSON contents of the config. But, sure, moving the path to the end will result in an easier-to-google error.

Reply:
> The value is a path within the tar archive, the JSON contents of the config.

**not** the JSON contents…